packages/root-cms/core/ai.ts

```diff
@@ -205,7 +205,12 @@ export class Chat {
     });
     // Using the `output` property provides both data and text responses.
     if (options.mode === 'edit') {
-      return res.output as AiResponse & {editData?: any};
+      const aiResponse = res.output as AiResponse & {editData?: any};
+      // Ensure a message is always provided, even if AI returns only data.
+      if (aiResponse?.data && !aiResponse?.message) {
+        aiResponse.message = 'Updated successfully.';
```

Comment:
i wonder if this might cause any false positives on our end, i.e. i wonder if we should just render a blank response or some other indication that there was no response instead of saying "updated successfully"

another thing might be to prompt the ai to always send some sort of confirmation message within the message field (or generally instruct it not to leave it blank).

Reply:
per chat i've removed the canned message, going to merge this now.